crates/sui-types/src/gas.rs

```diff
@@ -541,23 +556,23 @@ pub fn check_gas_balance(
 
     let max_gas_budget = cost_table.max_gas_budget as u128 * gas_price as u128;
     let min_gas_budget = cost_table.min_gas_budget_external() as u128 * gas_price as u128;
+    let required_gas_amount = gas_budget as u128;
 
-    if (gas_budget as u128) > max_gas_budget {
+    if required_gas_amount > max_gas_budget {
         return Err(UserInputError::GasBudgetTooHigh {
             gas_budget,
             max_budget: cost_table.max_gas_budget,
         });
     }
 
-    if (gas_budget as u128) < min_gas_budget {
+    if required_gas_amount < min_gas_budget {
         return Err(UserInputError::GasBudgetTooLow {
             gas_budget,
             min_budget: cost_table.min_gas_budget_external(),
         });
     }
 
     let mut gas_balance = get_gas_balance(gas_object)? as u128;
-    let required_gas_amount = (gas_budget as u128) * (gas_price as u128);
```

Comment:
this change is not gated and I am not sure it is an issue or not. Comments welcome here.
Basically the bug here was to fail signing of transactions with "not enough gas". The change corrects the check and the gas required is now the correct amount that is always lower that it used to be (we were multiplying a value and requiring the gas coin to be much more than needed).
In that sense every transaction that went through will still go through.
Execution is unaffected by this and so the results of the transaction did not change.
Is this ok?
@mystenmark @bmwill @longbowlu @amnn

Reply:
In general, it sounds okay to me (because it doesn't affect execution outcome, only signing).  But are you looking to land this in testnet before the 3.1 cut (i.e. during the fire drills next week?) If not, then the question of gating is moot anyway because we will regenesis.

Reply:
yes the idea is to redeploy kind of now ;)

Reply:
There is a risk to not gating. While rolling out the change, there will be a period of time when >f but <2f+1 validators have the new code. During this period of time, anyone who sends a tx that is signed only by either the old or the new code will have their objects locked, and will not be able to make a new tx to fix the problem.

If the tx passed on the old code, the objects will be locked until end of epoch. If it passes under the new code, the objects will be locked until the rollout finishes. If we think the rollout will be brief this may be acceptable.

The other risk is that there is some other issue related to byzantine signing behavior that I haven't thought of.

Reply:
Thankfully they're  aren't many txns happening today on this network